server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -2261,6 +2268,13 @@ public boolean expunge(UserVmVO vm, long callerUserId, Account caller) {
         }
         try {
 
+            if (vm.getBackupOfferingId() != null) {
+                List<Backup> backupsForVm = backupDao.listByVmId(vm.getDataCenterId(), vm.getId());
```

Comment:
@Pearl1594 I think we should also check for any backup policy and remove that too.